Release/v0.6.0

thomashebrard opened the pull request:
# Changelog

## [v0.6.0] - 2025-07-12

### Changed 
- The `Pipelex.make()` now takes a `relative_config_folder_path` argument, which is considered to be a relative path to the current working directory, pointing to the config folder path of pipelex. The argument `from_file` indicates if the path is relative to the caller's file path. If False, the path is considered to be absolute from where the process was started.

### Added
- Added github action for inference tests
- `load_json_list_from_path` function in `pipelex.tools.misc.file_utils`: Loads a JSON file and ensures it contains a list.

thomashebrard committed:
Refactor/improve architecture (#144)

* Better singleton

* Fixed major bug in identifying structure of implicit concept

---------

Co-authored-by: Louis Choquel <louis@pipelex.com>

thomashebrard committed:
Fix/ut concept library (#145)

thomashebrard committed:
add changelog v0.5.1

lchoquel committed:
Merge branch 'dev' of github.com:Pipelex/pipelex into dev

thomashebrard committed:
Fix/random (#148)

* fix/random

* push field

* don't use break line

* change manifesto link

* add pitch

* add changelog

thomashebrard committed:
fix/Pipelex config fixed path(#151)

thomashebrard committed:
Merge branch 'release/v0.6.0' into dev

thomashebrard committed:
dev to release/v0.6.0 (#152)

* Refactor/improve architecture (#144)

* Better singleton

* Fixed major bug in identifying structure of implicit concept

---------

Co-authored-by: Louis Choquel <louis@pipelex.com>

* Fix/ut concept library (#145)

* add changelog v0.5.1

* Fix/random (#148)

* fix/random

* push field

* don't use break line

* change manifesto link

* add pitch

* add changelog

* fix/Pipelex config fixed path(#151)

---------

Co-authored-by: Louis Choquel <louis@pipelex.com>

thomashebrard committed:
add branch main to manual-trigger-tests-check

thomashebrard committed:
Merge pull request #154 from Pipelex/dev

add branch main to manual-trigger-tests-check

thomashebrard committed:
Merge branch 'main' into release/v0.6.0

lchoquel commented on CHANGELOG.md:
```diff
@@ -1,5 +1,14 @@
+## [v0.6.0] - 2025-07-12
+
+### Changed 
+- The `Pipelex.make()` now takes a `relative_config_folder_path` argument, which is considered to be a relative path to the current working directory, pointing to the config folder path of pipelex. The argument `from_file` indicates if the path is relative to the caller's file path. If False, the path is considered to be absolute from where the process was started.
```
- `considered to be absolute from where the process was started` is confusing,  both cases are relative, right?
- this begs for an additional possibility of passing an absolute path

thomashebrard replied:
good point

lchoquel reviewed (changes requested):
remark on changelog, to fix 
+ feature request, can be for next release

thomashebrard committed:
absolte path in make

thomashebrard committed:
Merge branch 'release/v0.6.0' into dev

thomashebrard committed:
Merge pull request #155 from Pipelex/dev

absolte path in make

lchoquel committed:
Docs/improved issue templates (#170)

- improved issue templates
- updated Azure/OpenAI integrations, using dated deployment names systematically
- changed Makefile targets like `make doc` to `make docs` for consistency

lchoquel committed:
Merge branch 'dev' into release/v0.6.0

lchoquel committed:
Completed changelog

lchoquel committed:
Improved error handling in Pipelex.make() with meaningful error classes.

lchoquel reviewed (dismissed):
LGTM

lchoquel committed:
Release date
